Users/lukillgo/remove ruby

lkillgore opened the pull request:
Removed default installation of Ruby to use 'hosted tool cache' instead.

lkillgore commented on ubuntu/14.04/standard/Dockerfile:
```diff
@@ -154,10 +154,9 @@ RUN apt-get update \
 RUN apt-get update \
- && apt-get install -y ruby ruby-dev ruby-bundler \
- && rm -rf /var/lib/apt/lists/*
+ && apt-get install -y libz-dev openssl libssl-dev
```
These are necessary for the Ruby installation.  They were on the machine that was used to build the Ruby installation (these libraries were required for Ruby to access gem using https).  They are added here so that the Ruby installation in the tools cache will have access to them.

stepro commented:
@lkillgore, do you believe it is safe to remove the global ruby installation? No VSTS tools will attempt to use it? Also last I looked, the hosted tools cache only include Python; did this change?

lkillgore commented:
Hi @stepro, yes, we changed directions in our Ruby strategy.  I added Ruby recently--around~2-3 weeks ago--so no one should be using it.  And, yes, I added Ruby to the tools cache last week.  The way to have a definition use Ruby is to use a new vsts-task: 'Use Ruby Version', which will install off of the tools cache.

Installing a default version of ruby may cause unexpected collisions when the 'Use Ruby Version' task gets used.  Probably best to remove it, and funnel people to the task.  Thanks!

stepro commented:
The images have been updated.
